Sources_v3/StreamChatUI/ChatChannel/ChatChannelMessageComposerView.swift

```diff
@@ -4,165 +4,173 @@
 
 import UIKit
 
-open class ChatChannelMessageComposerView<ExtraData: UIExtraDataTypes>: UIInputView {
+open class ChatChannelMessageComposerView<ExtraData: UIExtraDataTypes>: UIInputView,
+    UIConfigProvider,
+    Customizable,
+    AppearanceSetting {
     // MARK: - Properties
     
-    public var buttonHeight: CGFloat = 20
+    public var attachmentsViewHeight: CGFloat = .zero
+    public var stateIconHeight: CGFloat = .zero
     
-    public let uiConfig: UIConfig<ExtraData>
-        
     // MARK: - Subviews
 
     public private(set) lazy var container = ContainerStackView().withoutAutoresizingMaskConstraints
     
     public private(set) lazy var replyView: UIView = .init()
     
-    public private(set) lazy var attachmentsView: MessageComposerAttachmentsView<ExtraData> = .init(uiConfig: uiConfig)
+    public private(set) lazy var attachmentsView: MessageComposerAttachmentsView<ExtraData> = {
+        uiConfig.messageComposer.attachmentsView.init().withoutAutoresizingMaskConstraints
+    }()
+    
+    public private(set) lazy var messageInputView: ChatChannelMessageInputView<ExtraData> = {
+        uiConfig.messageComposer.messageInputView.init().withoutAutoresizingMaskConstraints
+    }()
     
-    public private(set) lazy var messageInputView: ChatChannelMessageInputView<ExtraData> = .init(uiConfig: uiConfig)
+    public private(set) lazy var sendButton: MessageComposerSendButton<ExtraData> = {
+        uiConfig.messageComposer.sendButton.init().withoutAutoresizingMaskConstraints
+    }()
     
-    public private(set) lazy var sendButton: UIButton = {
-        let button = UIButton().withoutAutoresizingMaskConstraints
-        if #available(iOS 13.0, *) {
-            button.setImage(UIImage(systemName: "arrow.up.circle.fill"), for: .normal)
-        }
-        button.widthAnchor.constraint(equalTo: button.heightAnchor, multiplier: 1).isActive = true
+    public private(set) lazy var attachmentButton: UIButton = {
+        let button = uiConfig.messageComposer.composerButton.init().withoutAutoresizingMaskConstraints
+        let image = UIImage(named: "clip", in: .streamChatUI)?.tinted(with: uiConfig.colorPalette.messageComposerButton)
+        button.setImage(image, for: .normal)
```

Comment:
Isn't the `defaultAppearance` the right place for setting strings/images/colors?

Reply:
Yep, makes sense. It will be easy to change icon for button.